enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,225 @@
+
+- Multus supports overwriting cluster network definition by specifying a
+customized default net-attach-def in pod annotation. Customized net-attach-def
+will still use `ovn-k8s-cni-overlay` CNI plugin, but with an special annotation
+that requests a SR-IOV device.
+
+- RHCOS image shall be based on RHEL-8.3 for fully support of OVS Hardware
+Offload with connection track feature.
+
+- OpenvSwitch package version 2.13.
+
+- To use OVS Hardware Offload with link redundency, a bond interface needs to
+be created and used as geneve endpoint of OVS bridge. Mellanox SR-IOV VF-LAG
+provides hardware link-aggregation (LAG) for hardware offloaded ports. It
+allows offloading of OVS rules to a linux `bond` network device that combines
+two idential and hardware offload capable Physical Functions (PFs). With the
+SR-IOV VF LAG capability, the NIC PFs can receive the rules that the OVS tries
+to offload to the linux bond net device and offload them to the hardware
+e-switch.
+
+#### OVN-Kubernetes
+
+- OVS Hardware Offload is a [feature available](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/ovs_offload.md) in OVN-Kubernetes upstream.
+
+- `ovn-k8s-cni-overlay` CNI plugin configures the pod interface by moving the VF
+to pod namespace and plugging VF representor to `br-int` OVS bridge. The
+relevant [patches](https://github.com/openshift/ovn-kubernetes/commit/6c96467d0d3e58cab05641293d1c1b75e5914795) are available downstream.
+
+#### Cluster Network Operator
+
+- Cluster Network Operator will enable hardware offload and configure tc-policy
+on node OpenvSwitch instance and restart OpenvSwitch instance for hardware
+offload change to take effect.
+
+- Enabling `hw-offload` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:hw-offload=true`. The default value
+is `false`.
+
+- Setting `tc-policy` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:tc-policy=skip_sw`. The default
+value is `none`, optional values can be one of `none`, `skip_sw` and `skip_hw`.
+This field is only relevant if `hw-offload` is enabled.
+
+#### Multus
+
+- Multus will support specifying net-attach-def with customized annotations for
+[default cluster network in pod annotations](https://github.com/intel/multus-cni/blob/master/doc/configuration.md#specify-default-cluster-network-in-pod-annotations).
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
+[SriovNetworkNodePolicy API](https://github.com/openshift/sriov-network-operator/blob/master/pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go#L33).
+
+- SR-IOV config daemon (a DaemonSet managed by SR-IOV Operator) will support
+changing e-switch mode from `legacy` to `switchdev` on the PF device and
+exposing VF representor information in `SriovNetworkNodeState` status.
```

Comment:
Who will consume this information ? is it just for debug ?

Reply:
Yes, for debugging purpose, for example, to check whether switch dev mode is configured properly by looking at if VF has representor associated with it.